### ISSUE TITLE
Error updating store in subscription result handler

### ISSUE THREAD

**Issue by huforrest** [awslabs/aws-mobile-appsync-sdk-ios#101]
I subscribed to a new message subscription. When I get the new data and prepare to update the local store with the newly received data, I get an error. Here is the error log:

GraphQLResultError(path: ["allMessageConnection"], underlying: AWSAppSync.JSONDecodingError.missingValue)

And here is my code:

```swift
func startNewMessageSubscription() {
        let subscription = SubscribeToNewMessageSubscription(conversationId: conversationId)
        print("Starting new post subscription")
        do {
            newMessageSubscriptionWatcher = try appSyncClient?.subscribe(subscription: subscription, resultHandler: { (result, transaction, error) in
                print("Received new post!")
                if let result = result {
                    // Store a reference to the new object
                    let newMessage = result.data!.subscribeToNewMessage!
                    // Create a new object for the desired query where the new object content should reside
                    let messageToAdd = GetConversationMessagesQuery.Data.AllMessageConnection.Message(id: newMessage.id, conversationId: newMessage.conversationId, content: newMessage.content, createdAt: newMessage.createdAt, sender: newMessage.sender, isSent: newMessage.isSent)
                    do {
                        // Update the local store with the newly received data
                        try transaction?.update(query: GetConversationMessagesQuery(conversationId: newMessage.conversationId)) { (data: inout GetConversationMessagesQuery.Data) in
                            data.allMessageConnection?.messages?.append(messageToAdd)
                        }
                        self.loadAllMessagesFromCache(conversationId: newMessage.conversationId)
                    } catch let error {
                        print("Error updating store: \(error)")
                    }
                } else if let error = error {
                    print(error.localizedDescription)
                }
            })
        } catch {
            print("Error starting subscription.")
        }
    }
```

The error lies in this line:
data.allMessageConnection?.messages?.append(messageToAdd)

**Comment by palpatim** [comment]
This seems related to #92, at least based on the "missing value" message.

**Comment by palpatim** [comment]
@huforrest 

I'm writing a test to verify this behavior, which _might_ be fixed in #177. Can you confirm for me that the cache is empty at the time you're updating the subscription value? That is, you haven't yet processed the results of a `GetConversationMessagesQuery` from the server at the time you are updating the results of your subscription?

**Comment by palpatim** [comment]
@huforrest 

I've pushed a test to verify that the behavior I described above is handled by #177. If the behavior you're reporting in your original issue is different, please feel free to reopen this, but if not, I'll plan on closing this issue once we release 2.10.1 (which we're planning for later this week).

**Comment by palpatim** [comment]
This fix is released in 2.10.1. Please let us know if you continue to see issues.
